torchgen/gen_functionalization_type.py

```diff
@@ -395,14 +395,17 @@ def emit_view_functionalization_body(
           return {reverse_lambda.inner_call()}
         }}
       );
-      at::functionalization::impl::mutate_view_meta({view_tensor_name}, view_meta);
       {return_type} reference_tensor_output;
       {{
         at::AutoDispatchSkipFunctionalize guard;
         {meta_conversion_str}
         reference_tensor_output = at::_ops::{noop_api_name}::call({', '.join(meta_call_args)});
       }}
+      at::functionalization::impl::mutate_view_meta({view_tensor_name}, view_meta);
```

Comment:
For inplace_view ops, `mutate_view_meta()` is responsible for replaying the newly-added view, and will update the wrapper's size/stride metadata (e.g. with the new strides after running `transpose_()`

Before, I was running `mutate_view_meta()` **before** running the reference meta function, which caused it to run with the input meta tensors already having been mutated. The `reference_tensor_output`'s metadata was then wrong, so in `set_sizes_strides_offset` I'm setting the wrong size/stride info on the wrapper.

Reply:
These PR comments should just be comments in the code, they will be completely lost here.